### ISSUE TITLE
NoSuchMethodError is thrown from org.eclipse.compare.core

### ISSUE THREAD

**Issue by reprogrammer** [vazexqi/CodingSpectator#253]
There are multiple reports of the exceptions similar to the following in the error log of cs-103. This exception might not have anything to do with CodingTracker. But, since CodingTracker patches the `org.eclipse.compare` plug-in, I suggest that we make sure this problem is not due to CodingTracker.

<pre>
!MESSAGE Plug-in org.tigris.subversion.subclipse.ui was unable to load class org.tigris.subversion.subclipse.ui.decorator.SVNLightweightDecorator.
!STACK 0
org.eclipse.core.runtime.internal.adaptor.EclipseLazyStarter$TerminatingClassNotFoundException: An error occurred while automatically activating bundle org.tigris.subversion.subclipse.ui (568).
    at org.eclipse.core.runtime.internal.adaptor.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:121)
    at org.eclipse.osgi.baseadaptor.loader.ClasspathManager.findLocalClass(ClasspathManager.java:453)
    at org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader.findLocalClass(DefaultClassLoader.java:216)
    at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:393)
    at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:469)
    at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:422)
    at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:410)
    at org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader.loadClass(DefaultClassLoader.java:107)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:248)
    at org.eclipse.osgi.internal.loader.BundleLoader.loadClass(BundleLoader.java:338)
    at org.eclipse.osgi.framework.internal.core.BundleHost.loadClass(BundleHost.java:232)
    at org.eclipse.osgi.framework.internal.core.AbstractBundle.loadClass(AbstractBundle.java:1197)
    at org.eclipse.core.internal.registry.osgi.RegistryStrategyOSGI.createExecutableExtension(RegistryStrategyOSGI.java:174)
    at org.eclipse.core.internal.registry.ExtensionRegistry.createExecutableExtension(ExtensionRegistry.java:904)
    at org.eclipse.core.internal.registry.ConfigurationElement.createExecutableExtension(ConfigurationElement.java:243)
    at org.eclipse.core.internal.registry.ConfigurationElementHandle.createExecutableExtension(ConfigurationElementHandle.java:55)
    at org.eclipse.ui.internal.WorkbenchPlugin$1.run(WorkbenchPlugin.java:267)
    at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:52)
    at org.eclipse.ui.internal.WorkbenchPlugin.createExtension(WorkbenchPlugin.java:263)
    at org.eclipse.ui.internal.decorators.LightweightDecoratorDefinition$1.run(LightweightDecoratorDefinition.java:124)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.core.runtime.Platform.run(Platform.java:888)
    at org.eclipse.ui.internal.decorators.LightweightDecoratorDefinition.internalGetDecorator(LightweightDecoratorDefinition.java:120)
    at org.eclipse.ui.internal.decorators.LightweightDecoratorDefinition.decorate(LightweightDecoratorDefinition.java:251)
    at org.eclipse.ui.internal.decorators.LightweightDecoratorManager$LightweightRunnable.run(LightweightDecoratorManager.java:81)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.ui.internal.decorators.LightweightDecoratorManager.decorate(LightweightDecoratorManager.java:365)
    at org.eclipse.ui.internal.decorators.LightweightDecoratorManager.getDecorations(LightweightDecoratorManager.java:347)
    at org.eclipse.ui.internal.decorators.DecorationScheduler$1.ensureResultCached(DecorationScheduler.java:371)
    at org.eclipse.ui.internal.decorators.DecorationScheduler$1.run(DecorationScheduler.java:331)
    at org.eclipse.core.internal.jobs.Worker.run(Worker.java:54)
Caused by: org.osgi.framework.BundleException: Exception in org.tigris.subversion.subclipse.ui.SVNUIPlugin.start() of bundle org.tigris.subversion.subclipse.ui.
    at org.eclipse.osgi.framework.internal.core.BundleContextImpl.startActivator(BundleContextImpl.java:806)
    at org.eclipse.osgi.framework.internal.core.BundleContextImpl.start(BundleContextImpl.java:755)
    at org.eclipse.osgi.framework.internal.core.BundleHost.startWorker(BundleHost.java:370)
    at org.eclipse.osgi.framework.internal.core.AbstractBundle.start(AbstractBundle.java:284)
    at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:417)
    at org.eclipse.osgi.internal.loader.BundleLoader.setLazyTrigger(BundleLoader.java:265)
    at org.eclipse.core.runtime.internal.adaptor.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:106)
    ... 30 more
Caused by: java.lang.NoClassDefFoundError: org/eclipse/compare/IEncodedStreamContentAccessor
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClassCond(ClassLoader.java:632)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:616)
    at org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader.defineClass(DefaultClassLoader.java:188)
    at org.eclipse.osgi.baseadaptor.loader.ClasspathManager.defineClass(ClasspathManager.java:580)
    at org.eclipse.osgi.baseadaptor.loader.ClasspathManager.findClassImpl(ClasspathManager.java:550)
    at org.eclipse.osgi.baseadaptor.loader.ClasspathManager.findLocalClassImpl(ClasspathManager.java:481)
    at org.eclipse.osgi.baseadaptor.loader.ClasspathManager.findLocalClass_LockClassLoader(ClasspathManager.java:469)
    at org.eclipse.osgi.baseadaptor.loader.ClasspathManager.findLocalClass(ClasspathManager.java:449)
    at org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader.findLocalClass(DefaultClassLoader.java:216)
    at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:393)
    at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:469)
    at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:422)
    at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:410)
    at org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader.loadClass(DefaultClassLoader.java:107)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:248)
    at org.tigris.subversion.subclipse.ui.SVNUIPlugin.start(SVNUIPlugin.java:430)
    at org.eclipse.osgi.framework.internal.core.BundleContextImpl$1.run(BundleContextImpl.java:783)
    at java.security.AccessController.doPrivileged(Native Method)
    at org.eclipse.osgi.framework.internal.core.BundleContextImpl.startActivator(BundleContextImpl.java:774)
    ... 36 more
Caused by: org.eclipse.core.runtime.internal.adaptor.EclipseLazyStarter$TerminatingClassNotFoundException: An error occurred while automatically activating bundle org.eclipse.compare (338).
    at org.eclipse.core.runtime.internal.adaptor.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:121)
    at org.eclipse.osgi.baseadaptor.loader.ClasspathManager.findLocalClass(ClasspathManager.java:453)
    at org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader.findLocalClass(DefaultClassLoader.java:216)
    at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:393)
    at org.eclipse.osgi.internal.loader.SingleSourcePackage.loadClass(SingleSourcePackage.java:33)
    at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:466)
    at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:422)
    at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:410)
    at org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader.loadClass(DefaultClassLoader.java:107)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:248)
    ... 56 more
Caused by: org.osgi.framework.BundleException: Exception in org.eclipse.compare.internal.CompareUIPlugin.start() of bundle org.eclipse.compare.
    at org.eclipse.osgi.framework.internal.core.BundleContextImpl.startActivator(BundleContextImpl.java:806)
    at org.eclipse.osgi.framework.internal.core.BundleContextImpl.start(BundleContextImpl.java:755)
    at org.eclipse.osgi.framework.internal.core.BundleHost.startWorker(BundleHost.java:370)
    at org.eclipse.osgi.framework.internal.core.AbstractBundle.start(AbstractBundle.java:284)
    at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:417)
    at org.eclipse.osgi.internal.loader.BundleLoader.setLazyTrigger(BundleLoader.java:265)
    at org.eclipse.core.runtime.internal.adaptor.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:106)
    ... 65 more
Caused by: java.lang.NoSuchMethodError: org.eclipse.compare.internal.core.ComparePlugin.setCappingDisabled(Z)V
    at org.eclipse.compare.internal.CompareUIPlugin.start(CompareUIPlugin.java:261)
    at org.eclipse.osgi.framework.internal.core.BundleContextImpl$1.run(BundleContextImpl.java:783)
    at java.security.AccessController.doPrivileged(Native Method)
    at org.eclipse.osgi.framework.internal.core.BundleContextImpl.startActivator(BundleContextImpl.java:774)
    ... 71 more
</pre>

**Comment by Wanderer777** [comment]
@reprogrammer: What do you suggest to do to make sure that this is not caused by CodingTracker? The last time an exception related to `org.eclipse.compare` plugin was due to using an unsupported Eclipse version (Indigo).

**Comment by reprogrammer** [comment]
@Wanderer777: One way is to install Subclipse before and after CodingSpectator to see if you can reproduce the problem. You can ask the participant about the specific version of Subclipse that he/she has installed. Another way is to see if CodingTracker has added some dependency on `org.eclipse.compare.internal.core.ComparePlugin.setCappingDisabled`.

**Comment by reprogrammer** [comment]
@Wanderer777: It looks like [org.eclipse.compare.internal.core.ComparePlugin#setCappingDisabled](http://eclipsejavasource.appspot.com/vs?c=org.eclipse.compare.internal.core.ComparePlugin) existed in older versions of `org.eclipse.compare.core`. Does Subclipse come with an older version of `org.eclipse.compare.core` that gets patched by CodingTracker?

**Comment by reprogrammer** [comment]
@Wanderer777: `org.eclipse.compare.internal.core.ComparePlugin.setCappingDisabled(boolean)` exists in the version of `org.eclipse.compare` that CodingTracker patches for Eclipse Indigo. And, Eclipse Indigo shows the following error message in `org.eclipse.compare.internal.CompareUIPlugin.start(BundleContext)`.

<pre>
Discouraged access: The method setCappingDisabled(boolean) from the type ComparePlugin is not accessible due to restriction on required library /eclipse-rcp-indigo-linux-gtk-x86_64/plugins/org.eclipse.compare.core_3.5.200.I20110208-0800.jar
</pre>


Note that does not exist in the version of `org.eclipse.compare` that CodingTracker patches for Eclipse Helios. This makes me think that this issue is a Helios/Indigo problem.

**Comment by Wanderer777** [comment]
@reprogrammer: I do not think this problem is caused by CodingTracker due to the following (also, this answers your above questions):
- CodingTracker's code for Helios (at the time of this exception we supported Helios only) does not refer to anything from class `org.eclipse.compare.internal.core.ComparePlugin`;
- CodingTracker patches `org.eclipse.compare`, but does not patch `org.eclipse.compare.core` (though `org.eclipse.compare` depends on `org.eclipse.compare.core`), so it could not patch `org.eclipse.compare.core` of Subclipse (in case Subclipse comes with its own version of `org.eclipse.compare.core`);
- If it is the other way around, i.e. Subclipse is installed after CodingTracker and patches some plugins (in particular, `org.eclipse.compare`), the problem still could not be caused by CodingTracker, because it did not patch `org.eclipse.compare.core`.
- As you mentioned above, the call to `org.eclipse.compare.internal.core.ComparePlugin#setCappingDisabled` in the method `org.eclipse.compare.internal.CompareUIPlugin#start` appeared only in the Indigo's version of `org.eclipse.compare`, while at the time of this exception we supported Helios only.

So, I agree with your conclusion that this is a Helios/Indigo problem. And it was not caused by CodingTracker (even if CodingTracker was erroneously installed on Eclipse Indigo, it would not cause this particular problem). It looks like the user mixed some Indigo and Helios (or maybe even older) plugins in his/her Eclipse and this caused the above exception. In particular, it looks that the user used `org.eclipse.compare` from Indigo and `org.eclipse.compare.core` from an older version of Eclipse, in which this class did not have method `setCappingDisabled`.
